File: mixin-padang-webface/src/main/js/padang/model/XViewset.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import EReference from "webface/model/EReference";
import EAttribute from "webface/model/EAttribute";
import BasicEObject from "webface/model/BasicEObject";

import * as model from "padang/model/model";
import XMixture from "padang/model/XMixture";

export default class XViewset extends BasicEObject {

	public static XCLASSNAME: string = model.getEClassName("XViewset");

	public static FEATURE_MIXTURE = new EReference("mixture", XMixture);
	public static FEATURE_SELECTION = new EAttribute("selection", EAttribute.STRING);

	private mixture: XMixture = null;
	private selection: string = null;

	constructor() {
		super(model.createEClass(XViewset.XCLASSNAME), [
			XViewset.FEATURE_MIXTURE,
			XViewset.FEATURE_SELECTION,
		]);
	}

	public getMixture(): XMixture {
		return this.mixture;
	}

	public setMixture(newMixture: XMixture): void {
		let oldMixture = this.mixture;
		this.mixture = newMixture;
		this.eSetNotify(XViewset.FEATURE_MIXTURE, oldMixture, newMixture);
	}

	public getSelection(): string {
		return this.selection;
	}

	public setSelection(newSelection: string) {
		let oldSelection = this.selection;
		this.selection = newSelection;
		this.eSetNotify(XViewset.FEATURE_SELECTION, oldSelection, newSelection);
	}

}

File: mixin-padang/src/main/java/com/andia/mixin/padang/model/XViewset.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
package com.andia.mixin.padang.model;

import com.andia.mixin.model.BasicEObject;
import com.andia.mixin.model.EAttribute;
import com.andia.mixin.model.EFeature;
import com.andia.mixin.model.EReference;

public class XViewset extends BasicEObject {

	public static String XCLASSNAME = Padang.getEClassName("XViewset");

	public static EReference FEATURE_MIXTURE = new EReference("mixture", XMixture.class);
	public static EAttribute FEATURE_SELECTION = new EAttribute("selection", EAttribute.STRING);

	private XMixture mixture = null;
	private String selection = null;

	public XViewset() {
		super(Padang.createEClass(XCLASSNAME), new EFeature[] {
				XViewset.FEATURE_MIXTURE,
				XViewset.FEATURE_SELECTION,
		});
	}

	public XMixture getMixture() {
		return mixture;
	}

	public void setMixture(XMixture newMixture) {
		XMixture oldMixture = this.mixture;
		this.mixture = newMixture;
		this.eSetNotify(XViewset.FEATURE_MIXTURE, oldMixture, newMixture);
	}

	public String getSelection() {
		return this.selection;
	}

	public void setSelection(String newSelection) {
		String oldSelection = this.selection;
		this.selection = newSelection;
		this.eSetNotify(XViewset.FEATURE_SELECTION, oldSelection, newSelection);
	}

}
